### ISSUE TITLE
replace method

### ISSUE THREAD

**Issue by Adrien-B** [Rhinofly/ReactiveMongo-evolutions#3]
Being able to replace items value using their previous value.

Example instead of :

```
def valueFromId(x: Int): String = … 
id = rating[BSONInteger]("id")
newItem = item remove "id" add ("value" -> valueFromId(id.value))
```

Being able to write something like : 

```
def valueFrom(x:Int) = …
newItem = item replace ("value", valueFrom)
```

**Comment by EECOLOR** [comment]
Your suggested syntax does not include `"id"`. Using the current version you could use this:

```
def valueFromId(x: Int): String = … 
id = rating[Int]("id")
item update ("id" -> valueFromId(id)) rename ("id" -> "value")
```

My guess is that you would like to add the following signature to update:

```
def valueFromId(x: Int): String = … 
item update ("id" -> valueFromId) rename ("id" -> "value")
```

Is that correct?

**Comment by EECOLOR** [comment]
This is fixed in v0.3. The new syntax would be:

```
def valueFromId(x: Int): String = … 
item update ("id" -> valueFromId _)
```
